deployment/terraform/modules/osv/osv_api.tf

```diff
@@ -0,0 +1,116 @@
+# The OSV API on Cloud Run
+# Adapted from https://github.com/hashicorp/terraform-provider-google/issues/5528#issuecomment-1136040976
+
+data "google_container_registry_image" "api_backend" {
+  project = var.project_id
+  name    = "osv-server"
+  tag     = var.api_backend_image_tag
+}
+
+resource "google_cloud_run_service" "api_backend" {
+  project  = var.project_id
+  name     = "osv-grpc-backend"
+  location = "us-central1"
+
+  template {
+    spec {
+      containers {
+        image = data.google_container_registry_image.api_backend.image_url
+      }
+    }
+  }
+
+  traffic {
+    percent         = 100
+    latest_revision = true
+  }
+}
+
+resource "google_endpoints_service" "grpc_service" {
+  project      = var.project_id
+  service_name = var.api_url
+  grpc_config = templatefile(
+    "api/api_config.tftpl",
+    {
+      service_name = var.api_url,
+      backend_url  = replace(google_cloud_run_service.api_backend.status[0].url, "https://", "grpcs://")
+  })
+  protoc_output_base64 = filebase64("api/api_descriptor.pb")
```

Comment:
If api_descriptor.pb is changed, does this cause the proxy service to be re-deployed as well?

Reply:
Yep - those file functions stores the content of the file in the argument, so changes to the file will be picked up and propagated through to dependent resources.